client/src/api/v2/camera.js

```diff
@@ -39,3 +41,117 @@ export function useOverlays(device) {
 
   return { overlays, setActiveOverlay };
 }
+
+/**
+ * Send a message to server to start recording video
+ */
+export function startRecording() {
+  emit('start-camera-recording');
+}
+
+/**
+ * Send a message to server to stop recording video
+ */
+export function stopRecording() {
+  emit('stop-camera-recording');
+}
+
+/**
+ * @typedef {object} CameraRecordingStatusPayload
+ * @property {string} status Status of recording
+ * @property {?string} recordingMinutes Length of recording
+ * @property {?string} recordingFile File name of recording
+ * @property {?string} diskSpaceRemaining Remaining space on disk
+ * @property {?string} error Error message from camera
+ */
+
+/**
+ * Parse the payload into an object and format the values
+ *
+ * Payload structure is defined in the 'V3 MQTT Topics' page on Notion.
+ * Topic is /v3/camera/recording/status/<primary/>secondary>.
+ *
+ * @param {string} payload Payload
+ * @returns {CameraRecordingStatusPayload} Formatted payload without status
+ */
+function parsePayload(payload) {
+  const data = typeof payload === 'string' ? JSON.parse(payload) : payload;
+  if (!data) return null;
+
+  return Object.keys(data)
+    .reduce((acc, field) => {
+      let name = camelCaseToStartCase(field);
+      let value = '';
+
+      // format field value
+      switch (field) {
+        case 'status':
+          value = capitalise(data[field]);
+          break;
+
+        case 'diskSpaceRemaining':
+          value = formatBytes(data.diskSpaceRemaining);
+          break;
+
+        case 'recordingMinutes':
+          {
+            const mins = Math.floor(data.recordingMinutes);
+            const secs = Math.floor((data.recordingMinutes - mins) * 60);
+            value = `${mins}m ${secs}s`;
+            name = 'Recording Time';
+          }
+          break;
+
+        default:
+          value = data[field];
+          break;
+      }
+
+      acc[name] = value;
+      return acc;
+    }, {});
+}
+
+/**
+ * Initiate receiving camera recording statuses
+ */
+function initCameraStatus() {
+  emit('send-last-received-camera-recording-payloads');
+}
+
+/**
+ * Returns the last status payload published
+ *
+ * @param {string} device Device
+ * @returns {CameraRecordingStatusPayload} Payload
+ */
+export function useCameraRecordingStatus(device) {
+  // only run init once per render
+  useEffect(() => {
+    initCameraStatus();
+    return () => {};
```

Comment:
The leave callback is optional. You don't need to return anything if you don't use it.

Reply:
Done
4fcddfd